AdobeStockClient/Test/Unit/Model/Client/Files/CurlResultContainsEmptyFilesTest.php

```diff
@@ -0,0 +1,142 @@
+
+use Magento\AdobeIms\Model\Config as ImsConfig;
+use Magento\AdobeImsApi\Api\GetAccessTokenInterface;
+use Magento\AdobeStockClient\Model\Client\Files;
+use Magento\AdobeStockClient\Model\Config as ClientConfig;
+use Magento\Framework\Exception\IntegrationException;
+use Magento\Framework\HTTP\Client\Curl;
+use Magento\Framework\HTTP\Client\CurlFactory;
+use Magento\Framework\Locale\Resolver;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test case when the curl files requests returns and empty files data
+ */
+class CurlResultContainsEmptyFilesTest extends TestCase
+{
+    /**
+     * @var ImsConfig|MockObject
+     */
+    private $imsConfigMock;
+
+    /**
+     * @var ClientConfig|MockObject
+     */
+    private $clientConfigMock;
+
+    /**
+     * @var Resolver|MockObject
+     */
+    private $localeResolverMock;
+
+    /**
+     * @var GetAccessTokenInterface|MockObject
+     */
+    private $getAccessTokenMock;
+
+    /**
+     * @var Files|MockObject
+     */
+    private $files;
+
+    /**
+     * @var Curl|MockObject
+     */
+    private $curlAdapterMock;
+
+    /**
+     * @var Json|MockObject
+     */
+    private $jsonMock;
+
+    /**
+     * Prepare test objects.
+     */
+    protected function setUp(): void
+    {
+        $this->imsConfigMock = $this->createMock(ImsConfig::class);
+        $this->clientConfigMock = $this->createMock(ClientConfig::class);
+        $this->localeResolverMock = $this->createMock(Resolver::class);
+        $this->getAccessTokenMock = $this->createMock(GetAccessTokenInterface::class);
+        $this->jsonMock = $this->createMock(Json::class);
+        $curlFactoryMock = $this->createMock(CurlFactory::class);
+
+        $this->files = (new ObjectManager($this))->getObject(
+            Files::class,
+            [
+                'imsConfig' => $this->imsConfigMock,
+                'clientConfig' => $this->clientConfigMock,
+                'localeResolver' => $this->localeResolverMock,
+                'getAccessToken' => $this->getAccessTokenMock,
+                'curlFactory' => $curlFactoryMock,
+                'json' => $this->jsonMock,
+            ]
+        );
+
+        $this->curlAdapterMock = $this->createMock(Curl::class);
+        $curlFactoryMock->expects($this->once())
+            ->method('create')
+            ->willReturn($this->curlAdapterMock);
+    }
+
+    /**
+     * Search test
+     */
+    public function testResultContainsEmptyFiles(): void
+    {
+        $ids = [1];
+        $columns = [];
+        $locale = 'en';
+
+        $this->clientConfigMock->expects($this->once())
+            ->method('getProductName')
+            ->willReturn('string');
+
+        $this->imsConfigMock->expects($this->once())
+            ->method('getApiKey')
+            ->willReturn('string');
```

Comment:
Values used in unit tests should be assigned to variables. There are 2 possible situations:
1. Value is used verification of value flow - should be stored in a variable and used at least 2 times
2. Value is used as an argument to a method call - it is a good idea to provide such values from data provider as they can be used for boundary testing